pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -210,13 +246,22 @@ func (pr *PipelineRunStatus) GetCondition(t apis.ConditionType) *apis.Condition
 // InitializeConditions will set all conditions in pipelineRunCondSet to unknown for the PipelineRun
 // and set the started time to the current time
 func (pr *PipelineRunStatus) InitializeConditions() {
+	started := false
 	if pr.TaskRuns == nil {
 		pr.TaskRuns = make(map[string]*PipelineRunTaskRunStatus)
 	}
 	if pr.StartTime.IsZero() {
 		pr.StartTime = &metav1.Time{Time: time.Now()}
+		started = true
+	}
+	conditionManager := pipelineRunCondSet.Manage(pr)
+	conditionManager.InitializeConditions()
+	// Ensure the started reason is set for the "Succeeded" condition
+	if started {
+		initialCondition := conditionManager.GetCondition(apis.ConditionSucceeded)
+		initialCondition.Reason = TaskRunSucceededReasonStarted.String()
```

Comment:
Should this be `PipelineRunSucceededReasonStarted` instead of `TaskRunSucceededReasonStarted` ? 🤔  which makes me think, should we combine common PipelineRunReasons and TaskRunReasons e.g. `Started`, `Failed`, `Completed` etc?

Reply:
Nice catch, thank you!
I think I'd still keep them separated, even if the values are identical.